File: ios/Models/DWRUserProperties.m
//
//  DWRUserProperties.m
//  RNDeepWall
//
//  Created by Burak Yalcin on 10.11.2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

#import "DWRUserProperties.h"

@implementation DWRUserProperties

- (DeepWallUserProperties *)toDWObject {
	NSString *dwCountry = [DeepWallCountryManager getCountryByCode:self.country];
	NSString *dwLanguage = [DeepWallLanguageManager getLanguageByCode:self.language];
	
	DeepWallEnvironmentStyle dwEnvironmentStyle;
	if (self.environmentStyle != nil) {
		dwEnvironmentStyle = (DeepWallEnvironmentStyle)self.environmentStyle.integerValue;
	} else {
		dwEnvironmentStyle = DeepWallEnvironmentStyleAutomatic;
	}
	
    DeepWallUserProperties *dwUserProperties = [[DeepWallUserProperties alloc] initWithUuid:self.uuid country:dwCountry language:dwLanguage environmentStyle:dwEnvironmentStyle phoneNumber:self.phoneNumber debugAdvertiseAttributions:self.debugAdvertiseAttributions];
    dwUserProperties.emailAddress = self.emailAddress;
    dwUserProperties.firstName = self.firstName;
    dwUserProperties.lastName = self.lastName;
    return dwUserProperties;
}

@end

File: ios/Models/DWRUserProperties.swift
//
//  DWRUserProperties.swift
//  RNDeepWall
//
//  Created by Burak Yalcin on 8.09.2020.
//  Copyright © 2020 Facebook. All rights reserved.
//

import DeepWall

public final class DWRUserProperties: Decodable {
	
	public var uuid: String
	public var country: String
	public var language: String
	public var environmentStyle: UInt?
	public var debugAdvertiseAttributions: [String]?
}

extension DWRUserProperties {
	
	func toDWObject() -> DeepWallUserProperties {
		
		let dwCountry = DeepWallCountryManager.getCountry(by: self.country)
		let dwLanguage = DeepWallLanguageManager.getLanguage(by: self.language)
		let dwEnvironmentStyle: DeepWallEnvironmentStyle
		if let environmentStyle = self.environmentStyle {
            dwEnvironmentStyle = DeepWallEnvironmentStyle.init(rawValue: environmentStyle) ?? .automatic
		} else {
            dwEnvironmentStyle = .automatic
		}
		
		return DeepWallUserProperties(
			uuid: self.uuid,
			country: dwCountry,
			language: dwLanguage,
            environmentStyle: dwEnvironmentStyle,
			debugAdvertiseAttributions: self.debugAdvertiseAttributions)
	}
}
